### PR TITLE
Introduce Sonic RPC tests

### PR DESCRIPTION
**Pull request by jenikd**
This PR adds a RPC testing tool for checking, whether client contains needed RPC functions and if they respond with correct result.

This pipeline checkouts and starts fakenet client based on Sonic
Then it runs all the tests and shuts the node down

### CODE REVIEW

File: sonic/rpc-tests.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+                        sh 'echo "Start RPC tests"'
+
+                        dir('rpcTest') {
+                            sh 'cp .env.sample .env'
+                            sh 'go test -timeout 30s -run ^TestAllConfigs$ rpctesting/cmd/rpctesting -v -count=1'
+                        }
+                        
+                        sh 'echo "Stop fakenet"'
+                        sh 'pkill -f fakenet'
```
Comment (LuisPH3) [inline sonic/rpc-tests.jenkinsfile:98]: note, there is no process called fakenet.

Comment (Lubomir-Jahn) [inline sonic/rpc-tests.jenkinsfile:98]: Is this necessary? :)

Comment (jenikd) [reply sonic/rpc-tests.jenkinsfile:98]: It is not, but if sonicd is not stopped, then test will run till timeout. I can remove it and lower timeout value so resources are available sooner.
wdyt?

Comment (Lubomir-Jahn) [reply sonic/rpc-tests.jenkinsfile:98]: If this is the case, I agree with Wasuwee completely.

**Review by Lubomir-Jahn — APPROVED** [review]

**Pull request merged**